content/code-signing-yaml/signing-ios.md

```diff
@@ -0,0 +1,109 @@
+{{</notebox>}}
+
+## Automatic code signing
+
+In order to use automatic code signing and have Codemagic manage signing certificates and provisioning profiles on your behalf, you need to configure API access to App Store Connect.
+
+### Creating the App Store Connect API key
+
+It is recommended to create a dedicated App Store Connect API key for Codemagic in [App Store Connect](https://appstoreconnect.apple.com/access/api). To do so:
+
+1. Log in to App Store Connect and navigate to **Users and Access > Keys**.
+2. Click on the + sign to generate a new API key.
+3. Enter the name for the key and select an access level. We recommend choosing either `Developer` or `App Manager`, read more about Apple Developer Program role permissions [here](https://help.apple.com/app-store-connect/#/deve5f9a89d7).
+4. Click **Generate**.
+5. As soon as the key is generated, you can see it added in the list of active keys. Click **Download API Key** to save the private key for later. Note that the key can only be downloaded once.
+
+{{<notebox >}} 
+Take note of the **Issuer ID** above the table of active keys as well as the **Key ID** of the generated key as these will be required when setting up the Apple Developer Portal integration in Codemagic UI.
+{{</notebox>}}
+
+### Saving the API key to environment variables
+
+Save the API key and the related information as [environment](../getting-started/yaml#environment) variables. Make sure to [encrypt](../building/encrypting/#encrypting-sensitive-data) the values of the variables before adding them to the configuration file.
+
+    environment:
+      vars: 
+        APP_STORE_CONNECT_ISSUER_ID: Encrypted(...)
+        APP_STORE_CONNECT_KEY_IDENTIFIER: Encrypted(...)
+        APP_STORE_CONNECT_PRIVATE_KEY: Encrypted(...)
+        CERTIFICATE_PRIVATE_KEY: Encrypted(...)
+
+* `APP_STORE_CONNECT_KEY_IDENTIFIER`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Key ID** of the key.
+
+* `APP_STORE_CONNECT_ISSUER_ID`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Issuer ID** displayed above the table of active keys.
+
+* `APP_STORE_CONNECT_PRIVATE_KEY`
+
+  This is the private API key downloaded from App Store Connect. Note that when encrypting files via UI, they will be base64 encoded and would have to be decoded during the build. Alternativey, you can encrypt the **contents** of the file and save the encrypted value to the environment variable.
```

Comment:
Just a thought, but maybe to give an example how to decode, like
```
  app-store-connect fetch-signing-files "io.codemagic.app"
    --type IOS_APP_DEVELOPMENT
    --create
    --private-key $(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode)
```
or 
```
  echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode > decoded_private_key
  app-store-connect fetch-signing-files "io.codemagic.app"
    --type IOS_APP_DEVELOPMENT
    --create
    --private-key @file:decoded_private_key
```

But maybe to omit it right now and add a comprehensive section about yaml syntax and handling encryption and env vars and exit codes somewhere else, as it's applicable in other places as well.
I mean to emphasize that if it's a text file, user can encrypt just the content and won't need to decode it afterwards.

Reply:
I think it's a good idea to add that information under yaml syntax.